source/_components/sensor.dnsip.markdown

```diff
@@ -0,0 +1,50 @@
+---
+layout: page
+title: "DNS IP"
+description: "Instructions on how to integrate a DNS IP sensor into Home Assistant."
+date: 2017-02-25 11:05
+sidebar: true
+comments: false
+sharing: true
+footer: true
+ha_category: Sensor
+ha_iot_class: "Cloud Polling"
+ha_release: 0.40
+---
+
+
+The `dnsip` sensor will expose an IP address, fetched via DNS resolution, as its value. There are two operational modes:
+
+1. When you enable the sensor with minimal configuration, it will query [OpenDNS](https://www.opendns.com/)' nameserver with the URI `myip.opendns.com`, which will resolve to your external IP address. You can use this to get notified if your internet connection has been interrupted and you've been assigned a new IP address.
+2. If you specify a `hostname`, a regular DNS lookup will be performed, providing you the IP the hostname resolves to.
+
+You may also override the nameserver that's being used by setting the `resolver` parameter to any nameserver you like.
+
+To enable this sensor, add the following lines to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+sensor:
+  - platform: dnsip
+```
+
+Configuration variables:
+
+- **hostname** (*Optional*): The hostname for which to perform the DNS query.
+- **resolver** (*Optional*): The DNS server to target the query at.
+- **ipv6** (*Optional*): Set this to `true` or `false` if IPv6 should be used. Resolving the public IP will resolve to IP of the machine HASS is running on.
```

Comment:
The IP you'll get in that case it not the one of the modem/gateway, which may be responsible for port forwarding or provide some other services. If f. ex. there's an OpenVPN Server running and you want to be notified about the IP you would have to connect to from external networks, the IP this sensor would get probably wouln't be the one people need in case HASS is running on a different machine.